etcdmain/config_test.go

```diff
@@ -145,6 +141,60 @@ func TestConfigFileClusteringFields(t *testing.T) {
 	validateClusteringFlags(t, cfg)
 }
 
+func TestConfigFileClusteringFlags(t *testing.T) {
+	tests := []struct {
+		Name           string `json:"name"`
+		InitialCluster string `json:"initial-cluster"`
+		DNSCluster     string `json:"discovery-srv"`
+		Durl           string `json:"discovery"`
+	}{
+		{
+		// Use default name and generate a default inital-cluster
+		},
+		{
+			Name: "non-default",
+		},
+		{
+			InitialCluster: "0=localhost:8000",
+		},
+		{
+			Name:           "non-default",
+			InitialCluster: "0=localhost:8000",
+		},
+		{
+			DNSCluster: "example.com",
+		},
+		{
+			Name:       "non-default",
+			DNSCluster: "example.com",
+		},
+		{
+			Durl: "http://example.com/abc",
+		},
+		{
+			Name: "non-default",
+			Durl: "http://example.com/abc",
+		},
+	}
+
+	for i, tt := range tests {
+		b, err := yaml.Marshal(&tt)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		tmpfile := mustCreateCfgFile(t, b)
+		defer os.Remove(tmpfile.Name())
+
+		args := []string{fmt.Sprintf("--config-file=%s", tmpfile.Name())}
+
+		cfg := newConfig()
+		if err := cfg.parse(args); err != nil {
+			t.Errorf("%d: err = %v", i, err)
+		}
```

Comment:
It's not a parsing test, it's attempting to ensure that if independent discovery/bootstrap flags are correctly specified they *don't* falsely trigger an `embed.ErrConflictBootstrapFlags` error.

I'm assuming there are existing tests to validate parsing works?